Modules/xSharePoint/DSCResources/MSFT_xSPWebApplication/MSFT_xSPWebApplication.psm1

```diff
@@ -125,17 +151,17 @@ function Test-TargetResource
         [parameter(Mandatory = $false)] [System.String]  $Path,
         [parameter(Mandatory = $false)] [System.String]  $Port,
         [parameter(Mandatory = $false)] [System.Boolean] $UseSSL,
-        [parameter(Mandatory = $false)] [ValidateSet("NTLM","Kerberos")] [System.String] $AuthenticationMethod,
+        [parameter(Mandatory = $false)] [ValidateSet("NTLM","Kerberos")]  [System.String] $AuthenticationMethod,
+        [parameter(Mandatory = $false)] [ValidateSet("Present","Absent")] [System.String] $Ensure = "Present",
         [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
     )
 
     $CurrentValues = Get-TargetResource @PSBoundParameters
     Write-Verbose -Message "Testing for web application '$Name'"
-    if ($null -eq $CurrentValues) { return $false }
-
+    $PSBoundParameters.Ensure = $Ensure
     $testReturn = Test-xSharePointSpecificParameters -CurrentValues $CurrentValues `
                                                      -DesiredValues $PSBoundParameters `
-                                                     -ValuesToCheck @("ApplicationPool")
+                                                     -ValuesToCheck @("Ensure")
```

Comment:
Because it's the only thing we will change if it's not set correctly. When we add logic to the set method later to update other properties we can put them back in, but at this point it's the only thing that actually makes a difference if we want to call the set method.